funfact/lang/interpreter/_vectorize.py

```diff
@@ -84,8 +88,14 @@ def ein(
         self, lhs: P.Numeric, rhs: P.Numeric, precedence: int, reduction: str,
         pairwise: str, outidx: Optional[P.indices], live_indices, **kwargs
     ):
+        if self.post:
```

Comment:
Since the branching is strictly binary here, I think it is better to use `else` for the other return statement.

Reply:
And perhaps factor out the common `*[P.index...]` part and only handles the appending/prepending part in the branches?

Reply:
Done.